ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -988,7 +741,7 @@ private void releaseContext() {
         }
         context.clear(deleteResultDir);
         if (context.getHiveLocks() != null) {
-          hiveLocks.addAll(context.getHiveLocks());
+          driverTxnHandler.addHiveLocksFromContext();
```

Comment:
Do we need this magic? DriverTxnHandler already has a Context.
If I understand correctly this is just clearing the context at hand, should be the responsibility of the DriverTxnHandler.... or even better it can have a DriverTxnHandlerContext...

Reply:
The main problem here is that we have some HiveLocks in the hiveLocks variable (previously in Driver, after this patch in DriverTxnHandler), and also some HiveLocks in the Context. The two should be unified I believe, probably in Context. I think it's a good goal, but I'd rather not put it into this patch, as this is only moving code around. Let's do this in another jira, do you agree?

Reply:
Ok